packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,372 @@
+          [ChainId[BuiltInNetworkName.Mainnet]]: true,
+          [ChainId[BuiltInNetworkName.LineaMainnet]]: true,
+          [ChainId[BuiltInNetworkName.BaseMainnet]]: true,
+        },
+        solana: {
+          [SolScope.Mainnet]: true,
+        },
+      },
+    };
+  };
+
+// Metadata for the controller state
+const metadata = {
+  enabledNetworkMap: {
+    persist: true,
+    anonymous: true,
+  },
+};
+
+export class NetworkEnablementController extends BaseController<
+  typeof controllerName,
+  NetworkEnablementControllerState,
+  NetworkEnablementControllerMessenger
+> {
+  protected messagingSystem: NetworkEnablementControllerMessenger;
+
+  /**
+   * Creates a NetworkEnablementController instance.
+   *
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   */
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: NetworkEnablementControllerMessenger;
+    state?: Partial<NetworkEnablementControllerState>;
+  }) {
+    const defaultState = getDefaultNetworkEnablementControllerState();
+
+    // Deep merge enabledNetworkMap
+    const mergedEnabledNetworkMap = {
+      eip155: {
+        ...defaultState.enabledNetworkMap.eip155,
+        ...state?.enabledNetworkMap?.eip155,
+      },
+      solana: {
+        ...defaultState.enabledNetworkMap.solana,
+        ...state?.enabledNetworkMap?.solana,
+      },
+    };
+
+    // Call the constructor of BaseControllerV2
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { enabledNetworkMap: mergedEnabledNetworkMap },
+    });
+
+    this.messagingSystem = messenger;
+
+    // Subscribe to network state changes
+    this.messagingSystem.subscribe(
+      'NetworkController:networkAdded',
+      (networkAdded: NetworkConfiguration) => {
+        this.onNetworkAdded(networkAdded);
+      },
+    );
+
+    // Subscribe to network removal events
+    this.messagingSystem.subscribe(
+      'NetworkController:networkRemoved',
+      (networkRemoved: NetworkConfiguration) => {
+        this.onNetworkRemoved(networkRemoved);
+      },
+    );
+  }
+
+  /**
+   * Handles network added events.
+   * This method is called when a network is added to the network controller.
+   * The network is automatically enabled when added.
+   *
+   * @param networkAdded - The network added event.
+   */
+  private onNetworkAdded(networkAdded: NetworkConfiguration) {
+    this.setEvmEnabledNetwork(networkAdded.chainId);
+  }
+
+  /**
+   * Handles network removed events.
+   * This method is called when a network is removed from the network controller.
+   * The network is automatically disabled when removed.
+   *
+   * @param networkRemoved - The network removed event.
+   */
+  private onNetworkRemoved(networkRemoved: NetworkConfiguration) {
+    this.disableNetwork(networkRemoved.chainId);
+  }
```

Comment:
I think that we are already implementing this in the `ethereum-chain-utils.js` since that's where we have the currently `setTokenNetworkFilter` methods being called.

I'm open to handling this here, but just a heads up that this will need some additional modification on the extension.